### PR TITLE
Support multiple gateways in manifestwork deployer

### PR DESCRIPTION
**Pull request by mkolesnik**
Current implementation always overrides the existing ManifestWork even
if the requested MachineSet is different (as is the case for multiple
gateways).

In order to properly support multiple gateways, update the existing
ManifestWork only when its necessary to add a new MachineSet.

Fixes: https://github.com/stolostron/backlog/issues/23530
Fixes: https://github.com/stolostron/backlog/issues/23007

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

### CODE REVIEW

**Review by skitt — CHANGES_REQUESTED** [review]

File: pkg/cloud/manifestwork/machineset.go [hunk 1]
```diff
@@ -82,6 +97,35 @@ func (msd *manifestWorkMachineSetDeployer) Deploy(machineSet *unstructured.Unstr
...
+	// In case the machineset is already in the manifestwork, skip it not to cause disruptions
+	for _, manifest := range manifests {
+		existingMachineSet := &unstructured.Unstructured{}
+		if err := existingMachineSet.UnmarshalJSON(manifest.RawExtension.Raw); err != nil {
+			return err
+		}
+
+		if reflect.DeepEqual(machineSet, existingMachineSet) {
+			msd.eventRecorder.Eventf("ManifestWorkMachineSetDeployer",
+				"MachineSet %s already exists in ManifestWork %s", machineSet.GetName(), existingManifestWork.Name)
```
[the remark this answers is not in the document]
Comment (mkolesnik) [reply pkg/cloud/manifestwork/machineset.go:113]: Yes I was using it more as a log but you suggestion makes total sense, will push a fix promptly

**Pull request merged**